File: product/src/main/scala/de/leanovate/dose/product/model/ActiveProduct.scala
package de.leanovate.dose.product.model

import spray.json.DefaultJsonProtocol

import reactivemongo.bson.{BSONObjectID, Macros}
import de.leanovate.dose.product.util.TypeMapper._

case class ActiveProduct(
                          _id: Option[BSONObjectID],
                          name: String,
                          description: Option[String],
                          options: Seq[ProductOption],
                          categories: Seq[BSONObjectID],
                          images: Seq[ImageRef]
                          )

object ActiveProduct extends DefaultJsonProtocol {
  implicit val activeProductFormat = jsonFormat(ActiveProduct.apply, "id", "name", "description", "options", "categories", "images")

  implicit val activeProductHandler = Macros.handler[ActiveProduct]
}

File: billing/src/main/java/de/leanovate/dose/billing/model/ActiveProduct.java
package de.leanovate.dose.billing.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public class ActiveProduct {
    @JsonProperty
    public String id;

    @JsonProperty
    public String name;

    @JsonProperty
    public String description;

    @JsonProperty
    public List<ProductOption> options;

    @JsonProperty
    public List<String> categories;

    @JsonProperty
    public List<ImageRef> images;
}